web3/manager.py

```diff
@@ -117,6 +118,111 @@ def apply_null_result_formatters(
         return response
 
 
+def _validate_subscription_fields(response: RPCResponse) -> None:
+    params = response["params"]
+    subscription = params["subscription"]
+    if not isinstance(subscription, str) and not len(subscription) == 34:
+        _raise_bad_response_format(
+            response, "eth_subscription 'params' must include a 'subscription' field."
+        )
+
+
+def _validate_response(
+    response: RPCResponse,
+    error_formatters: Optional[Callable[..., Any]],
+    is_subscription_response: bool = False,
+    logger: Optional[logging.Logger] = None,
+) -> None:
+    if "jsonrpc" not in response or response["jsonrpc"] != "2.0":
+        _raise_bad_response_format(
+            response, 'The "jsonrpc" field must be present with a value of "2.0".'
```

Comment:
I know that this is part of the JSON-RPC spec, but if you haven't already, it may be worth a quick provider check to make sure that this actually does get sent back with most major providers. I can imagine some test providers or something not sending this (or something else we're validating) back.

Reply:
Tested against infura mainnet + sepolia, alchemy mainnet + sepolia, my local nethermind sepolia node, and polygon 👍🏼